openmetadata-ui/src/main/resources/ui/src/utils/DataQuality/DataQualityUtils.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright 2024 Collate.
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+import { TEST_CASE_FILTERS } from '../../constants/profiler.constant';
+import { ListTestCaseParamsBySearch } from '../../rest/testAPI';
+
+/**
+ * Builds the parameters for a test case search based on the given filters.
+ * @param params - The original test case parameters.
+ * @param filters - The filters to apply to the test case parameters.
+ * @returns The updated test case parameters with the applied filters.
+ */
+export const buildTestCaseParams = (
+  params: ListTestCaseParamsBySearch | undefined,
+  filters: string[]
+): ListTestCaseParamsBySearch => {
+  if (!params) {
+    return {};
+  }
+
+  const filterParams = (
+    paramKey: keyof ListTestCaseParamsBySearch,
+    filterKey: string
+  ) => (filters.includes(filterKey) ? { [paramKey]: params[paramKey] } : {});
+
+  return {
+    ...filterParams('endTimestamp', TEST_CASE_FILTERS.lastRun),
+    ...filterParams('startTimestamp', TEST_CASE_FILTERS.lastRun),
+    ...filterParams('entityLink', TEST_CASE_FILTERS.table),
+    ...filterParams('testPlatforms', TEST_CASE_FILTERS.platform),
+    ...filterParams('testCaseType', TEST_CASE_FILTERS.type),
+    ...filterParams('testCaseStatus', TEST_CASE_FILTERS.status),
+  };
+};
```

Comment:
Here we can create a map and iterate over it to avoid repetitive calls to `filterParams`.
```suggestion
export const buildTestCaseParams = (
  params: ListTestCaseParamsBySearch | undefined,
  filters: string[]
): ListTestCaseParamsBySearch => {
  if (!params) {
    return {};
  }

  const filterToParamKeyMap: Record<string, keyof ListTestCaseParamsBySearch> = {
    [TEST_CASE_FILTERS.lastRun]: 'endTimestamp',
    [TEST_CASE_FILTERS.lastRun]: 'startTimestamp',
    [TEST_CASE_FILTERS.table]: 'entityLink',
    [TEST_CASE_FILTERS.platform]: 'testPlatforms',
    [TEST_CASE_FILTERS.type]: 'testCaseType',
    [TEST_CASE_FILTERS.status]: 'testCaseStatus',
  };

  return filters.reduce((acc, filterKey) => {
    const paramKey = filterToParamKeyMap[filterKey];
    return paramKey ? { ...acc, [paramKey]: params[paramKey] } : acc;
  }, {});
};
```

Reply:
I think it will create issue for 
```
[TEST_CASE_FILTERS.lastRun]: 'endTimestamp',
[TEST_CASE_FILTERS.lastRun]: 'startTimestamp',
```
but I will give it a try

Reply:
With the approach described above, it's expected to return only a single key, which is `startTimestamp`, since it's the latest one.


https://github.com/open-metadata/OpenMetadata/assets/71748675/33b906eb-02eb-413b-afdc-6654b637d926